src/internal/m365/collection/site/restore_test.go

```diff
@@ -73,64 +74,90 @@ func (suite *SharePointRestoreSuite) TestListCollection_Restore() {
 	ctx, flush := tester.NewContext(t)
 	defer flush()
 
-	service := createTestService(t, suite.creds)
-	listing := spMock.ListDefault("Mock List")
-	testName := "MockListing"
-	listing.SetDisplayName(&testName)
-	byteArray, err := service.Serialize(listing)
+	testName, lrh, destName, mockData := suite.setupDependencies("genericList")
+
+	deets, err := restoreListItem(ctx, lrh, mockData, suite.siteID, destName)
 	require.NoError(t, err, clues.ToCore(err))
+	assert.Equal(t, fmt.Sprintf("%s_%s", destName, testName), deets.SharePoint.ItemName)
 
-	destName := testdata.DefaultRestoreConfig("").Location
+	// Clean-Up
+	suite.deleteList(ctx, lrh, t, deets)
+}
 
-	listData, err := data.NewPrefetchedItemWithInfo(
-		io.NopCloser(bytes.NewReader(byteArray)),
-		testName,
-		details.ItemInfo{SharePoint: ListToSPInfo(listing, int64(len(byteArray)))})
-	require.NoError(t, err, clues.ToCore(err))
+func (suite *SharePointRestoreSuite) TestListCollection_Restore_invalidTemplate() {
+	t := suite.T()
 
-	r, err := readers.NewVersionedRestoreReader(listData.ToReader())
-	require.NoError(t, err)
+	ctx, flush := tester.NewContext(t)
+	defer flush()
 
-	mockData := &dataMock.Item{
-		ItemID: testName,
-		Reader: r,
-	}
+	_, lrh, destName, mockData := suite.setupDependencies(api.WebTemplateExtensionsListTemplateName)
 
-	lrh := NewListsRestoreHandler(suite.siteID, suite.ac.Lists())
-	deets, err := restoreListItem(ctx, lrh, mockData, suite.siteID, destName)
-	require.NoError(t, err, clues.ToCore(err))
-	assert.Equal(t, fmt.Sprintf("%s_%s", destName, testName), deets.SharePoint.ItemName)
+	_, err := restoreListItem(ctx, lrh, mockData, suite.siteID, destName)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), api.ErrInvalidTemplateError.Error())
+}
 
-	// Clean-Up
+func (suite *SharePointRestoreSuite) deleteList(
+	ctx context.Context, lrh listsRestoreHandler, t *testing.T, deets details.ItemInfo,
+) {
```

Comment:
Per our code standards: func parameters should either be inlined or fully muti-lined, never a mix of the two.  Also, `t *testing.T` should always be the first param.

```suggestion
func (suite *SharePointRestoreSuite) deleteList(
	t *testing.T,
	ctx context.Context,
	lrh listsRestoreHandler,
	deets details.ItemInfo) {
```